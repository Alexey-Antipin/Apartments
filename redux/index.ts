import { checkbox, toogleBox, reset } from "./reducers/checkboxReducer";
import { newsThunk, countNewsThunk } from "./reducers/newsReducer";
import { amountOfRoomsThunk } from "./reducers/mainReducer";
import { articlesThunk } from "./reducers/articlesReducer";
import { choiceCity } from "./reducers/catalogReducer";
import {
  authorizationThunk,
  accountUser,
  accountDelete,
} from "./reducers/authorizationReducer";
import {
  redistrationThunk,
  confirmDataThunk,
  toggleCaptcha,
  captchaThunk,
} from "./reducers/registrationReducer";
import {
  contactsThunk,
  closeModal,
  openModal,
} from "./reducers/contactsReducer";
import {
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectCallCity,
  selectPriceMin,
  selectPriceMax,
  selectPlaces,
  defaultPrice,
  selectMetro,
  selectCity,
  selectArea,
} from "./reducers/selectReducer";

import { useAppDispatch, useAppSelector } from "../redux/hooks";
import type { RootState, AppDispatch } from "./store";

export {
  // Thunk
  authorizationThunk,
  amountOfRoomsThunk,
  redistrationThunk,
  confirmDataThunk,
  countNewsThunk,
  articlesThunk,
  contactsThunk,
  captchaThunk,
  newsThunk,

  // Method
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectPriceMin,
  selectPriceMax,
  selectCallCity,
  accountDelete,
  toggleCaptcha,
  selectPlaces,
  defaultPrice,
  selectMetro,
  accountUser,
  selectCity,
  selectArea,
  choiceCity,
  closeModal,
  toogleBox,
  openModal,
  checkbox,
  reset,

  // Hooks
  useAppDispatch,
  useAppSelector,

  //Type Store
  AppDispatch,
  RootState,
};
