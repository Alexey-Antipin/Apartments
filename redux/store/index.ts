import { configureStore } from "@reduxjs/toolkit";
import authorizationSlice from "../reducers/authorizationReducer";
import registrationSlice from "../reducers/registrationReducer";
import contactsSlice from "../reducers/contactsReducer";
import newsSlice from "../reducers/newsReducer";

export const store = configureStore({
  reducer: {
    authorization: authorizationSlice,
    registration: registrationSlice,
    contacts: contactsSlice,
    news: newsSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
