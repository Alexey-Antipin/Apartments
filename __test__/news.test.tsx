import { render } from "@testing-library/react";
import PaginatedPage from "../pages/news/";
import { Provider } from "react-redux";
import { store } from "../redux/store";

let articles: any[] = [];
let totalData: number = 0;
let currentPage: number = 0;

jest.mock("next/head", () => {
  return {
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }) => {
      return <>{children}</>;
    },
  };
});

describe("News", () => {
  test("Head", () => {
    render(
      <Provider store={store}>
        <PaginatedPage
          currentPage={currentPage}
          totalData={totalData}
          articles={articles}
        />
      </Provider>
    );
    expect(document.title).toBe("Новости");
  });
  test("Class", () => {
    const { getByRole } = render(
      <Provider store={store}>
        <PaginatedPage
          currentPage={currentPage}
          totalData={totalData}
          articles={articles}
        />
      </Provider>
    );
    let classButton = getByRole("button");
    expect(classButton).toHaveClass("button");
  });
  test("component snapshot", () => {
    const component = render(
      <Provider store={store}>
        <PaginatedPage
          currentPage={currentPage}
          totalData={totalData}
          articles={articles}
        />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
