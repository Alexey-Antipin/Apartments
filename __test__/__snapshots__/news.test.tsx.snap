// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <title>
          Новости
        </title>
        <div
          class="wrapper"
        >
          <div
            class="container"
          >
            <div>
              <div
                class="block"
              >
                <svg
                  fill="none"
                  height="12"
                  viewBox="0 0 12 12"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.7984 5.36427L6.41443 0.458394C6.17811 0.243027 5.82174 0.243051 5.58551 0.458371L0.201486 5.3643C0.0121814 5.5368 -0.0503497 5.80258 0.0421345 6.04138C0.134642 6.28019 0.359876 6.43448 0.615977 6.43448H1.4759V11.3498C1.4759 11.5447 1.63391 11.7027 1.8288 11.7027H4.7799C4.97478 11.7027 5.1328 11.5447 5.1328 11.3498V8.36537H6.86721V11.3498C6.86721 11.5447 7.02523 11.7027 7.22011 11.7027H10.1711C10.366 11.7027 10.524 11.5447 10.524 11.3498V6.43448H11.3841C11.6401 6.43448 11.8654 6.28016 11.9579 6.04138C12.0503 5.80256 11.9877 5.5368 11.7984 5.36427Z"
                    fill="#664EF9"
                  />
                </svg>
                <div
                  class="round"
                />
                <div
                  class="main option_v1_main"
                >
                  Новости
                </div>
              </div>
              <h2
                class="deepLink"
              >
                Новости
              </h2>
            </div>
          </div>
          <div
            class="container-list"
          >
            <ul
              class="link-articles"
            />
          </div>
          <div
            class="wrapper"
          />
          <div
            class="background"
          >
            <div
              class="block"
            >
              <input
                class="search"
                placeholder="Поиск по статьям"
                value=""
              />
              <button
                class="button"
              >
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 18 18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.8676 15.2581L13.5441 11.9344C13.3941 11.7844 13.1907 11.7011 12.9774 11.7011H12.434C13.3541 10.5243 13.9008 9.04417 13.9008 7.43401C13.9008 3.60364 10.7973 0.5 6.96711 0.5C3.13693 0.5 0.0334473 3.60364 0.0334473 7.43401C0.0334473 11.2644 3.13693 14.368 6.96711 14.368C8.57718 14.368 10.0573 13.8213 11.234 12.9012V13.4446C11.234 13.658 11.3173 13.8613 11.4673 14.0113L14.7908 17.335C15.1042 17.6483 15.6108 17.6483 15.9209 17.335L16.8642 16.3916C17.1776 16.0782 17.1776 15.5715 16.8676 15.2581ZM6.96711 11.7011C4.61033 11.7011 2.70024 9.79424 2.70024 7.43401C2.70024 5.07711 4.607 3.16693 6.96711 3.16693C9.32388 3.16693 11.234 5.07378 11.234 7.43401C11.234 9.79091 9.32722 11.7011 6.96711 11.7011Z"
                    fill="white"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <title>
        Новости
      </title>
      <div
        class="wrapper"
      >
        <div
          class="container"
        >
          <div>
            <div
              class="block"
            >
              <svg
                fill="none"
                height="12"
                viewBox="0 0 12 12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.7984 5.36427L6.41443 0.458394C6.17811 0.243027 5.82174 0.243051 5.58551 0.458371L0.201486 5.3643C0.0121814 5.5368 -0.0503497 5.80258 0.0421345 6.04138C0.134642 6.28019 0.359876 6.43448 0.615977 6.43448H1.4759V11.3498C1.4759 11.5447 1.63391 11.7027 1.8288 11.7027H4.7799C4.97478 11.7027 5.1328 11.5447 5.1328 11.3498V8.36537H6.86721V11.3498C6.86721 11.5447 7.02523 11.7027 7.22011 11.7027H10.1711C10.366 11.7027 10.524 11.5447 10.524 11.3498V6.43448H11.3841C11.6401 6.43448 11.8654 6.28016 11.9579 6.04138C12.0503 5.80256 11.9877 5.5368 11.7984 5.36427Z"
                  fill="#664EF9"
                />
              </svg>
              <div
                class="round"
              />
              <div
                class="main option_v1_main"
              >
                Новости
              </div>
            </div>
            <h2
              class="deepLink"
            >
              Новости
            </h2>
          </div>
        </div>
        <div
          class="container-list"
        >
          <ul
            class="link-articles"
          />
        </div>
        <div
          class="wrapper"
        />
        <div
          class="background"
        >
          <div
            class="block"
          >
            <input
              class="search"
              placeholder="Поиск по статьям"
              value=""
            />
            <button
              class="button"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 18 18"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.8676 15.2581L13.5441 11.9344C13.3941 11.7844 13.1907 11.7011 12.9774 11.7011H12.434C13.3541 10.5243 13.9008 9.04417 13.9008 7.43401C13.9008 3.60364 10.7973 0.5 6.96711 0.5C3.13693 0.5 0.0334473 3.60364 0.0334473 7.43401C0.0334473 11.2644 3.13693 14.368 6.96711 14.368C8.57718 14.368 10.0573 13.8213 11.234 12.9012V13.4446C11.234 13.658 11.3173 13.8613 11.4673 14.0113L14.7908 17.335C15.1042 17.6483 15.6108 17.6483 15.9209 17.335L16.8642 16.3916C17.1776 16.0782 17.1776 15.5715 16.8676 15.2581ZM6.96711 11.7011C4.61033 11.7011 2.70024 9.79424 2.70024 7.43401C2.70024 5.07711 4.607 3.16693 6.96711 3.16693C9.32388 3.16693 11.234 5.07378 11.234 7.43401C11.234 9.79091 9.32722 11.7011 6.96711 11.7011Z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
